### PR TITLE
Adds `rd_kafka_header_length` to public api

### PR DESCRIPTION
**Pull request by johnistan**

### CODE REVIEW

File: src/rdkafka.h [hunk 1]
```diff
@@ -1014,6 +1014,12 @@ rd_kafka_header_iter_all (const rd_kafka_headers_t *hdrs, size_t idx,
                           const char **namep,
                           const void **valuep, size_t *sizep);
 
+/**
+ * @brief Returns the number of header key/value pairs
+ *
+ * @param hdrs   Headers to count
+ */
+RD_EXPORT int rd_kafka_header_length (const rd_kafka_headers_t *hdrs);
```
Comment (edenhill) [inline src/rdkafka.h]: Plural `headers_length`, or `header_cnt`, to avoid ambiguity (number of headers rather than length of one header)

Comment (edenhill) [inline src/rdkafka.h]: Should return a `size_t`

File: tests/0072-headers_ut.c [hunk 2]
```diff
@@ -261,27 +261,52 @@ static rd_kafka_resp_err_t on_send1 (rd_kafka_t *rk,
                 "hey"
         };
         rd_kafka_headers_t *hdrs;
+        int header_length;
         rd_kafka_resp_err_t err;
         const void *value;
         size_t size;
 
+
```
Comment (edenhill) [inline tests/0072-headers_ut.c]: remove whitespace diff

File: tests/0072-headers_ut.c [hunk 3]
```diff
@@ -261,27 +261,52 @@ static rd_kafka_resp_err_t on_send1 (rd_kafka_t *rk,
...
         expect_check(__FUNCTION__, expected, rkmessage);
 
         err = rd_kafka_message_headers(rkmessage, &hdrs);
+
```
Comment (edenhill) [inline tests/0072-headers_ut.c]: remove whitespace diff

**Comment by johnistan** [comment]
closed in favor of #1614

**Pull request closed without merging**